src/test/java/DataAndPropertyProvider.java

```diff
@@ -0,0 +1,72 @@
+import org.apache.commons.io.FileUtils;
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+import org.testng.annotations.DataProvider;
+
+import java.io.File;
+import java.io.IOException;
+
+public class DataAndPropertyProvider {
+
+    public static Object[][] readFile(File file) throws IOException {
+        String text = null;
+        Logger log = LogManager.getRootLogger();
+
+        try {
+            text = FileUtils.readFileToString(file);
+        } catch (IOException e) {
+            log.error(e.getMessage());
+        }
+        String[] row = text.split(";");
+
+        int rowNum = row.length;
+        int colNum = row[0].split(",").length;
+
+        Object[][] data = new Integer[rowNum][colNum];
+
+        for (int i = 0; i < rowNum; i++) {
+            String[] cols = row[i].split(",");
+
+            for (int j = 0; j < colNum; j++) {
+                String string = cols[j].trim();
+                data[i][j] = Integer.valueOf(string);
+                System.out.println("value is " + data[i][j]);
+            }
+        }
+        return data;
+    }
+
+    private static Object[][] getDataSets(String filePath){
+        Object[][] returnObjArray = null;
+        File file = new File(filePath);
+        try {
+            returnObjArray = readFile(file);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return returnObjArray;
+    }
+
+    @DataProvider(name = "IncorrectDataSets")
+    public static Object[][] incorrectDataSets() {
+        Object[][] returnObjArray = new Object[0][];
+        getDataSets("src/test/resources/IncorrectDataSets.txt");
+        return returnObjArray;
```

Comment:
Just return empty array?

Reply:
Optional: just "return getDataSets("src/test/resources/IncorrectDataSets.txt");", there is no needs to create variable and then return it.